### PR TITLE
KAA-913 Bucket delivery timeout causes sequential switching of access points

### PR DESCRIPTION
**Pull request by forGGe**
…s points

### CODE REVIEW

File: client/client-multi/client-cpp/impl/log/LogCollector.cpp [hunk 1]
```diff
@@ -86,11 +86,16 @@ void LogCollector::processTimeout()
     KAA_MUTEX_UNIQUE_DECLARE(timeoutsGuardLock, timeoutsGuard_);
     KAA_MUTEX_LOCKED("timeoutsGuard_");
 
-    for (const auto& request : timeouts_) {
-        storage_->rollbackBucket(request.first);
-    }
+    auto now = clock_t::now();
 
-    timeouts_.clear();
+    for (auto request = timeouts_.begin();request!=timeouts_.end();) {
```
Comment (rasendubi) [inline client/client-multi/client-cpp/impl/log/LogCollector.cpp:91]: code style

File: client/client-multi/client-cpp/impl/log/LogCollector.cpp [hunk 2]
```diff
@@ -86,11 +86,16 @@ void LogCollector::processTimeout()
...
+        if (now >= request->second.getTimeoutTime()) {
+            storage_->rollbackBucket(request->first);
+            timeouts_.erase(request++);
```
Comment (rasendubi) [inline client/client-multi/client-cpp/impl/log/LogCollector.cpp:94]: Move `++request` to for loop and use `request` here.

Comment (rasendubi) [inline client/client-multi/client-cpp/impl/log/LogCollector.cpp:94]: Maybe use for-each

File: client/client-multi/client-cpp/impl/log/LogCollector.cpp [hunk 3]
```diff
@@ -371,6 +376,13 @@ void LogCollector::switchAccessPoint()
             KAA_LOG_WARN("Try to switch to another Operations server...");
             channelManager_->onServerFailed(logChannel->getServer(),
                                             KaaFailoverReason::OPERATION_SERVERS_NA);
+            KAA_MUTEX_LOCKING("timeoutsGuard_");
+            KAA_MUTEX_UNIQUE_DECLARE(timeoutsGuardLock, timeoutsGuard_);
+            KAA_MUTEX_LOCKED("timeoutsGuard_");
+            for( const auto &request : timeouts_) {
```
Comment (rasendubi) [inline client/client-multi/client-cpp/impl/log/LogCollector.cpp:382]: code style

File: client/client-multi/client-cpp/test/headers/log/MockLogUploadStrategy.hpp [hunk 4]
```diff
@@ -29,7 +29,7 @@ class MockLogUploadStrategy: public ILogUploadStrategy {
 
     virtual std::size_t getTimeout() { ++onGetTimeout_; return timeout_; }
 
-    virtual void onTimeout(ILogFailoverCommand& controller) { ++onTimeout_; }
+    virtual void onTimeout(ILogFailoverCommand& controller) { ++onTimeout_; controller.switchAccessPoint(); }
```
Comment (rasendubi) [inline client/client-multi/client-cpp/test/headers/log/MockLogUploadStrategy.hpp:32]: code style

**Comment by forGGe** [comment]
Pr closing: another one must be opened from original repository

**Pull request closed without merging**